added main file. Bug fix. File renaming

yicheng0922 opened the pull request:
I did some bug fix and created the main file. The result of the optimization is strange that first of all, there are positive values in the result vector which indicate that some springs actually extended. Second, con_func returns a positive value with the result vector passed in which indicates that the equality constraint is not satisfied.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/edrumwri/optimized-compliance/6)
<!-- Reviewable:end -->

yicheng0922 committed:
added main file. Bug fix. File renaming

edrumwri commented:
Reviewed 9 of 10 files at r1.
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[matlab/unit_test_find_closest_vector_sector_box.m, line 4 at r1](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-KXbpSqxmYANab0c_u-_:-KXbpSqxmYANab0c_u-a:bndjqqz) ([raw file](https://github.com/edrumwri/optimized-compliance/blob/3135eb65aae733c01bdd94160bc08e4223b84eb2/matlab/unit_test_find_closest_vector_sector_box.m#L4)):*
> ```objc
> shape1.x = 3
> shape1.y = 3
> shape1.w = 1
> ```

Why did you change these values in the unit test?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6)*
<!-- Sent from Reviewable.io -->

yicheng0922 commented:
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[matlab/unit_test_find_closest_vector_sector_box.m, line 4 at r1](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-KXbpSqxmYANab0c_u-_:-KXc_gKV0jZB152HAg8D:b-lowhv7) ([raw file](https://github.com/edrumwri/optimized-compliance/blob/3135eb65aae733c01bdd94160bc08e4223b84eb2/matlab/unit_test_find_closest_vector_sector_box.m#L4)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Why did you change these values in the unit test?
</blockquote></details>

I changed the data structure of the box such that x and y represents the midpoint of the bottom edge instead of the center of the box. The old data were for the old data structure so I changed it so that it is now using the current data structure.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
Reviewed 1 of 10 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[matlab/main.m, line 31 at r1](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-KXcbHuZaadDiP2jayRn:-KXcbHuZaadDiP2jayRo:b5xybhd) ([raw file](https://github.com/edrumwri/optimized-compliance/blob/3135eb65aae733c01bdd94160bc08e4223b84eb2/matlab/main.m#L31)):*
> ```objc
> 
> % compute the spring compression 
> X = fmincon(@(x) (obj_func(x, box1_spring_num, box2_spring_num, box1_spring_k,box2_spring_k)), zeros(total_spring_num,1)*0,[],[],[],[],[],[],@(x) con_func(x,box1_spring,box2_spring),options);
> ```

Nice use of anonymous functions here. May I suggest the code I was using instead?

% setup the objective and constraint functions
ofn = @(x) obj_func(x, box1_springs, box2_springs, k1, k2);
cfn = @(x) con_func(x, box1_springs, box2_springs);

 % compute the spring lengths (for forty-three springs in the example) 
X   = fmincon(ofn, zeros(total_spring_num,1),[],[],[],[],[],[],cfn,options);

    

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
I'm suggesting a change for readability below. Then <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-:-KXcbkWpb6lGmU5aG8wy:bglobik)*
<!-- Sent from Reviewable.io -->

yicheng0922 committed:
update main for readability

edrumwri commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6)*
<!-- Sent from Reviewable.io -->

yicheng0922 committed:
modified comment

yicheng0922 commented:
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[matlab/main.m, line 31 at r1](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-KXcbHuZaadDiP2jayRn:-KXcrE2heCyTCZryIpW4:b-896fix) ([raw file](https://github.com/edrumwri/optimized-compliance/blob/3135eb65aae733c01bdd94160bc08e4223b84eb2/matlab/main.m#L31)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Nice use of anonymous functions here. May I suggest the code I was using instead?

% setup the objective and constraint functions
ofn = @(x) obj_func(x, box1_springs, box2_springs, k1, k2);
cfn = @(x) con_func(x, box1_springs, box2_springs);

 % compute the spring lengths (for forty-three springs in the example) 
X   = fmincon(ofn, zeros(total_spring_num,1),[],[],[],[],[],[],cfn,options);

    
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 8 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-:-KXcrsO0-ZY3pyfFT4xB:bnfp4nl)*
<!-- Sent from Reviewable.io -->

yicheng0922 commented:
Hi, I made a change in the comment so that the total spring number in the comment actually match the spring number. Could you check it so that I can merge it? Reviewable is not letting me merge because it thinks that change is not reviewed

---

Review status: 8 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-:-KXcsEjvD-LIW9eNLPqP:b-ep6rfu)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 8 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6#-:-KXct1NznwrHsjwWtPzS:bnfp4nl)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/edrumwri/optimized-compliance/6)*
<!-- Sent from Reviewable.io -->
